helix-vte/src/lib.rs

```diff
@@ -0,0 +1,180 @@
+pub mod error;
+
+use std::{
+    collections::HashMap,
+    path::PathBuf,
+    pin::Pin,
+    sync::{
+        atomic::{AtomicU32, Ordering},
+        Arc,
+    },
+};
+
+use bytes::Bytes;
+use error::Error;
+use futures::{
+    select,
+    stream::{self, SelectAll},
+    FutureExt, Stream, StreamExt,
+};
+use pty_process::{OwnedWritePty, Pty, Size};
+use tokio::{io::AsyncWriteExt, sync::Notify};
+use tokio_util::io::ReaderStream;
+
+pub type TerminalId = u32;
+
+static TERMINAL_ID_SEQ: AtomicU32 = AtomicU32::new(0);
+
+#[derive(Debug, Clone)]
+pub enum PtyMessage {
+    Input(Vec<u8>),
+    Resize(u16, u16),
+    Close,
+}
+
+#[derive(Debug, Clone)]
+pub enum PtyEvent {
+    Data(Bytes),
+    Error(String),
+    Terminated(i32),
+}
+
+pub struct PtySpawnConfig {
+    pub command: String,
+    pub arguments: Option<Vec<String>>,
+    pub size: Option<(u16, u16)>,
+    pub cwd: Option<PathBuf>,
+    pub env: Option<HashMap<String, String>>,
+}
+
+impl Default for PtySpawnConfig {
+    fn default() -> Self {
+        Self {
+            command: "/usr/bin/zsh".to_string(),
```

Comment:
Let's pass this in and use `editor.config.shell` https://github.com/helix-editor/helix/blob/5835d0b4601e29ed4001db0c1ad365f02d88d9e3/helix-view/src/editor.rs#L124-L125

Reply:
I think we shouldn't use that option, because by default it is just `/bin/sh` and that is good for simple cases like pipe through some program. But when we starting the terminal we usually would like to see default system shell program. For now I am setting it value from `SHELL` env variable, but later it would be better make some terminal section in main configuration file.